### DISCUSSION TITLE
Direct Messaging Key Pair Problem After Configuration Import

### DISCUSSION THREAD

**Discussion by f16flcn** [meshtastic/Discussions#97] [General]
Hi Community!

I manage a network of around 2-3 dozen Meshtastic devices, configured as clients, repeaters, and routers. These devices are provisioned using a standard configuration files. However, I've noticed the following issue:

Problem with Key Pairs:

When importing the configuration file, the security key pair (private and public) is included in the import file.
As a result, all devices now share the same key pair, causing direct messaging encryption to fail (since all devices use the same keys).
Key Sync Issue:

After upgrading the firmware, even if I generate new keys, the devices do not sync the updated public keys with the network.
I suspect the key synchronization is only triggered when the node name (node ID) changes, but I’m unsure if this is correct or if there’s another trigger for key resync.

Also via the Android App, it does not seem to be possible to set the key pair to blank and reboot the device to generate new keys (as the documentation suggests - maybe only via CLI?).

Questions:
Is there a way to regenerate or resync key pairs across all nodes without:

1. Re-flashing firmware?
Changing node names (do not believe the Node ID is configurable)?
Completely re-configuring the devices manually?
Can I forget the device and will that resync?

2. What triggers key synchronization between nodes, and how can I force a re-sync of public keys for the affected devices (I understand that there a trust implication with this)?

3. Is there a way to modify the configuration import process to exclude the key pairs and allow each device to generate unique keys?

Thank you for any assistance or insights!

**Comment by lupusworax** [comment]
Same problem, any solution?

**Reply by madeofstown** [reply]
If you are using the python CLI to do the backup/restores then it should be as easy as opening the backup file with a text editor and removing the lines that set the keys.

**Reply by lupusworax** [reply]
> If you are using the python CLI to do the backup/restores then it should be as easy as opening the backup file with a text editor and removing the lines that set the keys.

Thanks I simply had to erase.uf2 it several times (1 time did not get rid of it for some reason and yes I did open the serial command line to actually erase it ;) ) and then after a fresh install/ flash it worked.

**Reply by fluffrider** [reply]
Having the same issue. I have not used python before. Not opposed to trying it. But will the radios sync on their own, given enough time "talking" to each other?

**Reply by lupusworax** [reply]
> Having the same issue. I have not used python before. Not opposed to trying it. But will the radios sync on their own, given enough time "talking" to each other?

if you happen to be a beta tester of the android app, opt out of it and reinstall the stable meshtastic version, see if it works now.

**Comment by fluffrider** [comment]
I am using the IOS On Apr 23, 2025, at 1:48 PM, Loup ***@***.***> wrote:﻿

Having the same issue. I have not used python before. Not opposed to trying it. But will the radios sync on their own, given enough time "talking" to each other?

if you happen to be a beta tester of the android app, opt out of it and reinstall the stable meshtastic version, see if it works now.

**Reply by garthvh** [reply]
Did you do a full config import and change your key or are you just seeing some key mismatches?

**Comment by fluffrider** [comment]
I did a full reboot flash, and yes I stupidly didn’t think to back the keys up because I didn’t have any discreet channels set up. I was under the impression that it’d just work normally. I can send over the public channel and to some of my nodes. I am not sure what happened, but it seems like this node corrupted another of my nodes because I can’t talk to it with my tdeck now.(the initial issues was a heltec and my peripheral network are Rak nodes) My Seeed tracker card can talk to the corrupted node and my tdeck, so I am really confused.On Apr 23, 2025, at 6:42 PM, Garth Vander Houwen ***@***.***> wrote:﻿
Did you do a full config import and change your key or are you just seeing some key mismatches?
